### ISSUE TITLE
Neo4j return incorrect query result

### ISSUE THREAD

**Issue by kikimo** [neo4j/neo4j#12995]
Look at the queries below. The first query return no result while the second query which change the last where predicate `(((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0))))` of the first query into its return result, all the returned rows shows that this predicate is evaluated to be true which conflict to the first query:

```txt
$ MATCH (v0)-[e0:Rel_0]->()-[e1]->(v1) WHERE (v0.id in [30, 7, 11, 29, 16, 6, 2, 4, 19, 27, 11]) AND (v1.Label_0_3_Bool) WITH (- e1.Rel_0_4_Double) AS pa0 WHERE ((pa0 = pa0) AND (((pa0 + 0.490657) <= pa0) OR (NOT (NOT (pa0 <= pa0))))) WITH (- ((pa0 * (((- pa0) - 0.810811) * (- (pa0 * pa0)))) / (56 % pa0))) AS pa0, (- pa0) AS pa1, (pa0 < pa0) AS pa2, (0.988976 + ((pa0 * pa0) % pa0)) AS pa3, (pa0 < pa0) AS pa4 WHERE (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0)))) RETURN count(*) as totalRows

╒═══════════╕
│"totalRows"│
╞═══════════╡
│0          │
└───────────┘

$ MATCH (v0)-[e0:Rel_0]->()-[e1]->(v1) WHERE (v0.id in [30, 7, 11, 29, 16, 6, 2, 4, 19, 27, 11]) AND (v1.Label_0_3_Bool) WITH (- e1.Rel_0_4_Double) AS pa0 WHERE ((pa0 = pa0) AND (((pa0 + 0.490657) <= pa0) OR (NOT (NOT (pa0 <= pa0))))) WITH (- ((pa0 * (((- pa0) - 0.810811) * (- (pa0 * pa0)))) / (56 % pa0))) AS pa0, (- pa0) AS pa1, (pa0 < pa0) AS pa2, (0.988976 + ((pa0 * pa0) % pa0)) AS pa3, (pa0 < pa0) AS pa4 return (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0))))

╒══════════════════════════════════════════════════════════════════════╕
│"(((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa│
│1 % pa0) - 0.777156)) * (- pa0))))"                                   │
╞══════════════════════════════════════════════════════════════════════╡
│true                                                                  │
├──────────────────────────────────────────────────────────────────────┤
│true                                                                  │
├──────────────────────────────────────────────────────────────────────┤
│true                                                                  │
└──────────────────────────────────────────────────────────────────────┘
```

Neo4j version in test: neo4j-community-5.1.0

@HannesSandberg @arnefischereit @inmost-light PTAL

**Comment by arnefischereit** [comment]
Hi @kikimo,

Could you please also detail the dataset that you run these queries on? Also, it would be great if you could include the plans that you obtain via prefpending [`EXPLAIN`](https://neo4j.com/docs/cypher-manual/current/query-tuning/#how-do-i-profile-a-query) to the query.

**Comment by kikimo** [comment]
@arnefischereit 

plan for 
```txt
explain MATCH (v0)-[e0:Rel_0]->()-[e1]->(v1) WHERE (v0.id in [30, 7, 11, 29, 16, 6, 2, 4, 19, 27, 11]) AND (v1.Label_0_3_Bool) WITH (- e1.Rel_0_4_Double) AS pa0 WHERE ((pa0 = pa0) AND (((pa0 + 0.490657) <= pa0) OR (NOT (NOT (pa0 <= pa0))))) WITH (- ((pa0 * (((- pa0) - 0.810811) * (- (pa0 * pa0)))) / (56 % pa0))) AS pa0, (- pa0) AS pa1, (pa0 < pa0) AS pa2, (0.988976 + ((pa0 * pa0) % pa0)) AS pa3, (pa0 < pa0) AS pa4 WHERE (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0)))) RETURN count(*) as totalRows
```

![plan](https://user-images.githubusercontent.com/451943/205601068-5e5aa3d1-f182-41bd-acf6-76e73d12d825.svg)

plan for 
```txt
explain MATCH (v0)-[e0:Rel_0]->()-[e1]->(v1) WHERE (v0.id in [30, 7, 11, 29, 16, 6, 2, 4, 19, 27, 11]) AND (v1.Label_0_3_Bool) WITH (- e1.Rel_0_4_Double) AS pa0 WHERE ((pa0 = pa0) AND (((pa0 + 0.490657) <= pa0) OR (NOT (NOT (pa0 <= pa0))))) WITH (- ((pa0 * (((- pa0) - 0.810811) * (- (pa0 * pa0)))) / (56 % pa0))) AS pa0, (- pa0) AS pa1, (pa0 < pa0) AS pa2, (0.988976 + ((pa0 * pa0) % pa0)) AS pa3, (pa0 < pa0) AS pa4 return (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0))))
```

![plan (7)](https://user-images.githubusercontent.com/451943/205601374-b93a4acc-388e-486a-afca-2444bd9746ec.svg)

**Comment by kikimo** [comment]
@arnefischereit and the data set in use:

[n4j.txt](https://github.com/neo4j/neo4j/files/10152730/n4j.txt)

**Comment by arnefischereit** [comment]
Thank you for the nice script to set things up.

I imported the data but unfortunately all versions which I try do not return results for query 2:

```
neo4j@test> PROFILE MATCH (v0)-[e0:Rel_0]->()-[e1]->(v1) WHERE (v0.id in [30, 7, 11, 29, 16, 6, 2, 4, 19, 27, 11]) AND (v1.Label_0_3_Bool) WITH (- e1.Rel_0_4_Double) AS pa0 WHERE ((pa0 = pa0) AND (((pa0 + 0.490657) <= pa0) OR (NOT (NOT (pa0 <= pa0))))) WITH (- ((pa0 * (((- pa0) - 0.810811) * (- (pa0 * pa0)))) / (56 % pa0))) AS pa0, (- pa0) AS pa1, (pa0 < pa0) AS pa2, (0.988976 + ((pa0 * pa0) % pa0)) AS pa3, (pa0 < pa0) AS pa4 return (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0))))
            ;
+---------------------------------------------------------------------------------------------------------+
| (((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0)))) |
+---------------------------------------------------------------------------------------------------------+
+---------------------------------------------------------------------------------------------------------+
```

But the plan also looks different:

```
+-------------------------------+----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| Operator                      | Id | Details                                                                                              | Estimated Rows | Rows | DB Hits | Page Cache Hits/Misses |
+-------------------------------+----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +ProduceResults               |  0 | `(((- pa1) + (- (pa3 / (- 86)))) < (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0 |              0 |    0 |       0 |                    0/0 |
| |                             |    | ))))`                                                                                                |                |      |         |                        |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Projection                   |  1 | ($autoint_10 - pa1) + ($autoint_11 - pa3 / $autoint_12) < $autoint_13 - (((pa1 + pa3) - ($autoint_14 |              0 |    0 |       0 |                    0/0 |
| |                             |    | - pa3)) + (pa1 % pa0 - $autodouble_15)) * ($autoint_16 - pa0) AS `(((- pa1) + (- (pa3 / (- 86)))) <  |                |      |         |                        |
| |                             |    | (- ((((pa1 + pa3) - (- pa3)) + ((pa1 % pa0) - 0.777156)) * (- pa0))))`                               |                |      |         |                        |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Projection                   |  2 | $autodouble_9 + (pa0 * pa0) % pa0 AS pa3, pa0 < pa0 AS pa2, $autoint_3 - (pa0 * ((($autoint_4 - pa0) |              0 |    0 |       0 |                    0/0 |
| |                             |    | - $autodouble_5) * ($autoint_6 - pa0 * pa0))) / ($autoint_7 % pa0) AS pa0, $autoint_8 - pa0 AS pa1,  |                |      |         |                        |
| |                             |    | pa0 < pa0 AS pa4                                                                                     |                |      |         |                        |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Filter                       |  3 | pa0 = pa0 AND (pa0 + $autodouble_2 <= pa0 OR pa0 <= pa0)                                             |              0 |    0 |       0 |                    0/0 |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Projection                   |  4 | $autoint_1 - e1.Rel_0_4_Double AS pa0                                                                |              0 |    0 |       0 |                    0/0 |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Filter                       |  5 | v0.id IN $autolist_0 AND not e1 = e0                                                                 |              0 |    0 |       0 |                    0/0 |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Expand(All)                  |  6 | (anon_0)<-[e0:Rel_0]-(v0)                                                                            |              1 |    0 |       0 |                    0/0 |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +Filter                       |  7 | CoerceToPredicate(v1.Label_0_3_Bool)                                                                 |              2 |    0 |       0 |                    0/0 |
| |                             +----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
| +DirectedAllRelationshipsScan |  8 | (anon_0)-[e1]->(v1)                                                                                  |              3 |    0 |       0 |                    0/0 |
+-------------------------------+----+------------------------------------------------------------------------------------------------------+----------------+------+---------+------------------------+
```

That makes me think that your graph does not quite look like described in the file you attached.

Could you run `CALL  db.stats.retrieve("GRAPH COUNTS");` and provide the output of that?

**Comment by kikimo** [comment]
Hi @arnefischereit , this is the output of ` CALL db.stats.retrieve("GRAPH COUNTS");` from my Neo4j:

```txt
$ CALL db.stats.retrieve("GRAPH COUNTS");

╒══════════════╤══════════════════════════════════════════════════════════════════════╕
│"section"     │"data"                                                                │
╞══════════════╪══════════════════════════════════════════════════════════════════════╡
│"GRAPH COUNTS"│{"relationships":[{"count":100},{"relationshipType":"Rel_3","count":16│
│              │},{"relationshipType":"Rel_3","startLabel":"Label_4","count":6},{"rela│
│              │tionshipType":"Rel_3","count":6,"endLabel":"Label_4"},{"relationshipTy│
│              │pe":"Rel_3","startLabel":"Label_0","count":10},{"relationshipType":"Re│
│              │l_3","count":8,"endLabel":"Label_0"},{"relationshipType":"Rel_3","star│
│              │tLabel":"Label_1","count":5},{"relationshipType":"Rel_3","count":7,"en│
│              │dLabel":"Label_1"},{"relationshipType":"Rel_3","startLabel":"Label_3",│
│              │"count":5},{"relationshipType":"Rel_3","count":7,"endLabel":"Label_3"}│
│              │,{"relationshipType":"Rel_3","startLabel":"ID","count":16},{"relations│
│              │hipType":"Rel_3","count":16,"endLabel":"ID"},{"relationshipType":"Rel_│
│              │3","startLabel":"Label_6","count":13},{"relationshipType":"Rel_3","cou│
│              │nt":10,"endLabel":"Label_6"},{"relationshipType":"Rel_3","startLabel":│
│              │"Label_5","count":8},{"relationshipType":"Rel_3","count":7,"endLabel":│
│              │"Label_5"},{"relationshipType":"Rel_3","startLabel":"Label_2","count":│
│              │6},{"relationshipType":"Rel_3","count":6,"endLabel":"Label_2"},{"relat│
│              │ionshipType":"Rel_1","count":15},{"relationshipType":"Rel_1","startLab│
│              │el":"Label_4","count":6},{"relationshipType":"Rel_1","count":7,"endLab│
│              │el":"Label_4"},{"relationshipType":"Rel_1","startLabel":"Label_0","cou│
│              │nt":9},{"relationshipType":"Rel_1","count":7,"endLabel":"Label_0"},{"r│
│              │elationshipType":"Rel_1","startLabel":"Label_1","count":7},{"relations│
│              │hipType":"Rel_1","count":4,"endLabel":"Label_1"},{"relationshipType":"│
│              │Rel_1","startLabel":"Label_3","count":6},{"relationshipType":"Rel_1","│
│              │count":5,"endLabel":"Label_3"},{"relationshipType":"Rel_1","startLabel│
│              │":"ID","count":15},{"relationshipType":"Rel_1","count":15,"endLabel":"│
│              │ID"},{"relationshipType":"Rel_1","startLabel":"Label_6","count":10},{"│
│              │relationshipType":"Rel_1","count":11,"endLabel":"Label_6"},{"relations│
│              │hipType":"Rel_1","startLabel":"Label_5","count":8},{"relationshipType"│
│              │:"Rel_1","count":7,"endLabel":"Label_5"},{"relationshipType":"Rel_1","│
│              │startLabel":"Label_2","count":5},{"relationshipType":"Rel_1","count":7│
│              │,"endLabel":"Label_2"},{"relationshipType":"Rel_2","count":24},{"relat│
│              │ionshipType":"Rel_2","startLabel":"Label_4","count":11},{"relationship│
│              │Type":"Rel_2","count":9,"endLabel":"Label_4"},{"relationshipType":"Rel│
│              │_2","startLabel":"Label_0","count":12},{"relationshipType":"Rel_2","co│
│              │unt":14,"endLabel":"Label_0"},{"relationshipType":"Rel_2","startLabel"│
│              │:"Label_1","count":11},{"relationshipType":"Rel_2","count":8,"endLabel│
│              │":"Label_1"},{"relationshipType":"Rel_2","startLabel":"Label_3","count│
│              │":8},{"relationshipType":"Rel_2","count":10,"endLabel":"Label_3"},{"re│
│              │lationshipType":"Rel_2","startLabel":"ID","count":24},{"relationshipTy│
│              │pe":"Rel_2","count":24,"endLabel":"ID"},{"relationshipType":"Rel_2","s│
│              │tartLabel":"Label_6","count":19},{"relationshipType":"Rel_2","count":1│
│              │6,"endLabel":"Label_6"},{"relationshipType":"Rel_2","startLabel":"Labe│
│              │l_5","count":11},{"relationshipType":"Rel_2","count":15,"endLabel":"La│
│              │bel_5"},{"relationshipType":"Rel_2","startLabel":"Label_2","count":7},│
│              │{"relationshipType":"Rel_2","count":9,"endLabel":"Label_2"},{"relation│
│              │shipType":"Rel_0","count":18},{"relationshipType":"Rel_0","startLabel"│
│              │:"Label_4","count":7},{"relationshipType":"Rel_0","count":6,"endLabel"│
│              │:"Label_4"},{"relationshipType":"Rel_0","startLabel":"Label_0","count"│
│              │:10},{"relationshipType":"Rel_0","count":8,"endLabel":"Label_0"},{"rel│
│              │ationshipType":"Rel_0","startLabel":"Label_1","count":5},{"relationshi│
│              │pType":"Rel_0","count":6,"endLabel":"Label_1"},{"relationshipType":"Re│
│              │l_0","startLabel":"Label_3","count":6},{"relationshipType":"Rel_0","co│
│              │unt":9,"endLabel":"Label_3"},{"relationshipType":"Rel_0","startLabel":│
│              │"ID","count":18},{"relationshipType":"Rel_0","count":18,"endLabel":"ID│
│              │"},{"relationshipType":"Rel_0","startLabel":"Label_6","count":13},{"re│
│              │lationshipType":"Rel_0","count":11,"endLabel":"Label_6"},{"relationshi│
│              │pType":"Rel_0","startLabel":"Label_5","count":11},{"relationshipType":│
│              │"Rel_0","count":9,"endLabel":"Label_5"},{"relationshipType":"Rel_0","s│
│              │tartLabel":"Label_2","count":6},{"relationshipType":"Rel_0","count":6,│
│              │"endLabel":"Label_2"},{"relationshipType":"Rel_5","count":15},{"relati│
│              │onshipType":"Rel_5","startLabel":"Label_4","count":6},{"relationshipTy│
│              │pe":"Rel_5","count":5,"endLabel":"Label_4"},{"relationshipType":"Rel_5│
│              │","startLabel":"Label_0","count":8},{"relationshipType":"Rel_5","count│
│              │":6,"endLabel":"Label_0"},{"relationshipType":"Rel_5","startLabel":"La│
│              │bel_1","count":5},{"relationshipType":"Rel_5","count":7,"endLabel":"La│
│              │bel_1"},{"relationshipType":"Rel_5","startLabel":"Label_3","count":5},│
│              │{"relationshipType":"Rel_5","count":7,"endLabel":"Label_3"},{"relation│
│              │shipType":"Rel_5","startLabel":"ID","count":15},{"relationshipType":"R│
│              │el_5","count":15,"endLabel":"ID"},{"relationshipType":"Rel_5","startLa│
│              │bel":"Label_6","count":12},{"relationshipType":"Rel_5","count":10,"end│
│              │Label":"Label_6"},{"relationshipType":"Rel_5","startLabel":"Label_5","│
│              │count":9},{"relationshipType":"Rel_5","count":7,"endLabel":"Label_5"},│
│              │{"relationshipType":"Rel_5","startLabel":"Label_2","count":5},{"relati│
│              │onshipType":"Rel_5","count":6,"endLabel":"Label_2"},{"relationshipType│
│              │":"Rel_4","count":12},{"relationshipType":"Rel_4","startLabel":"Label_│
│              │4","count":4},{"relationshipType":"Rel_4","count":5,"endLabel":"Label_│
│              │4"},{"relationshipType":"Rel_4","startLabel":"Label_0","count":6},{"re│
│              │lationshipType":"Rel_4","count":5,"endLabel":"Label_0"},{"relationship│
│              │Type":"Rel_4","startLabel":"Label_1","count":4},{"relationshipType":"R│
│              │el_4","count":3,"endLabel":"Label_1"},{"relationshipType":"Rel_4","sta│
│              │rtLabel":"Label_3","count":3},{"relationshipType":"Rel_4","count":4,"e│
│              │ndLabel":"Label_3"},{"relationshipType":"Rel_4","startLabel":"ID","cou│
│              │nt":12},{"relationshipType":"Rel_4","count":12,"endLabel":"ID"},{"rela│
│              │tionshipType":"Rel_4","startLabel":"Label_6","count":9},{"relationship│
│              │Type":"Rel_4","count":9,"endLabel":"Label_6"},{"relationshipType":"Rel│
│              │_4","startLabel":"Label_5","count":5},{"relationshipType":"Rel_4","cou│
│              │nt":6,"endLabel":"Label_5"},{"relationshipType":"Rel_4","startLabel":"│
│              │Label_2","count":4},{"relationshipType":"Rel_4","count":6,"endLabel":"│
│              │Label_2"},{"relationshipType":"Rel_11","count":0},{"relationshipType":│
│              │"Rel_13","count":0},{"relationshipType":"Rel_8","count":0},{"relations│
│              │hipType":"Rel_12","count":0},{"relationshipType":"Rel_14","count":0},{│
│              │"relationshipType":"Rel_6","count":0},{"relationshipType":"Rel_10","co│
│              │unt":0},{"relationshipType":"Rel_7","count":0},{"relationshipType":"Re│
│              │l_9","count":0},{"relationshipType":"like","count":0},{"relationshipTy│
│              │pe":"serve","count":0},{"relationshipType":"teammate","count":0}],"nod│
│              │es":[{"count":31},{"count":14,"label":"Label_4"},{"count":15,"label":"│
│              │Label_0"},{"count":12,"label":"Label_1"},{"count":13,"label":"Label_3"│
│              │},{"count":31,"label":"ID"},{"count":19,"label":"Label_6"},{"count":17│
│              │,"label":"Label_5"},{"count":11,"label":"Label_2"},{"count":0,"label":│
│              │"Label_7"},{"count":0,"label":"Label_12"},{"count":0,"label":"Label_10│
│              │"},{"count":0,"label":"Label_9"},{"count":0,"label":"Label_11"},{"coun│
│              │t":0,"label":"Label_8"},{"count":0,"label":"player"},{"count":0,"label│
│              │":"team"},{"count":0,"label":"bachelor"}],"indexes":[{"updatesSinceEst│
│              │imation":0,"indexType":"BTREE","totalSize":14,"properties":["Label_4_0│
│              │_Int","Label_4_1_String"],"labels":["Label_4"],"estimatedUniqueSize":1│
│              │4},{"updatesSinceEstimation":0,"indexType":"BTREE","totalSize":31,"pro│
│              │perties":["id"],"labels":["ID"],"estimatedUniqueSize":31},{"updatesSin│
│              │ceEstimation":0,"indexType":"LOOKUP","totalSize":0,"properties":[],"la│
│              │bels":[],"estimatedUniqueSize":0},{"updatesSinceEstimation":0,"indexTy│
│              │pe":"BTREE","totalSize":15,"relationshipTypes":["Rel_1"],"properties":│
│              │["Rel_1_4_Bool","Rel_1_3_Double","Rel_1_5_Int"],"estimatedUniqueSize":│
│              │15},{"updatesSinceEstimation":0,"indexType":"BTREE","totalSize":16,"re│
│              │lationshipTypes":["Rel_3"],"properties":["Rel_3_2_Double"],"estimatedU│
│              │niqueSize":16},{"updatesSinceEstimation":0,"indexType":"BTREE","totalS│
│              │ize":15,"properties":["Label_0_2_String"],"labels":["Label_0"],"estima│
│              │tedUniqueSize":14},{"updatesSinceEstimation":0,"indexType":"BTREE","to│
│              │talSize":12,"properties":["Label_1_1_String","Label_1_0_String"],"labe│
│              │ls":["Label_1"],"estimatedUniqueSize":12},{"updatesSinceEstimation":0,│
│              │"indexType":"BTREE","totalSize":13,"properties":["Label_3_0_Bool","Lab│
│              │el_3_5_Int","Label_3_1_String"],"labels":["Label_3"],"estimatedUniqueS│
│              │ize":13},{"updatesSinceEstimation":0,"indexType":"LOOKUP","totalSize":│
│              │0,"relationshipTypes":[],"properties":[],"estimatedUniqueSize":0}],"co│
│              │nstraints":[]}                                                        │
└──────────────┴──────────────────────────────────────────────────────────────────────┘
```

**Comment by kikimo** [comment]
Hi @arnefischereit , my apology, the Neo4j I'm using is neo4j-community-4.4.14 instead of neo4j-community-5.1.0.
Perhaps fiexed in  neo4j-community-5.1.0?

**Comment by arnefischereit** [comment]
Thank you, @kikimo for sending me the graph counts. That way I could see that the import script had failed on my machine and I could now reproduce the issue.

**Comment by kikimo** [comment]
@arnefischereit Hi, I just upload the whole Noe4j instance I'm runing to google drive, you can try download it: https://drive.google.com/file/d/1Zpq3ACZmiCpCkTDF5jgY-KoSPfy98d7l/view?usp=share_link

**Comment by kikimo** [comment]
> Thank you, @kikimo for sending me the graph counts. That way I could see that the import script had failed on my machine and I could now reproduce the issue.

@arnefischereit I misunderstood you and thought that you were not able to reproduce the issue so I provide you the google drive download link:-)

**Comment by gem-neo4j** [comment]
Hello, thank you for reporting :) This is now fixed and will be released in 5.13 and 4.4.27
